### ISSUE TITLE
How to use my own table data to train the Model

### ISSUE THREAD

**Issue by Sharathmk99** [rinkstiekema/PDF-Table-Structure-Recognition-using-deep-learning#12]
Hi,

This is an amazing project. Thank you for open sourcing the same.

I'm trying to extract table data for invoice Images. Can you help me how can i train the model with my data?

Thank you in advance.

**Comment by rinkstiekema** [comment]
Thank you for your interest in this project. 
First of all, I'd like to advice you to use another repository for anything but experimental research. The project is not much more than a proof of concept, performing OK but nothing flashy is to be expected at this stage. 

There may be specialized, pretrained, models for your case. It highly depends on the nature of your data. Consider the following: 

1. Are the invoices of digital nature or are they scanned documents?
2. Do the invoices (roughly) follow the same template, or is there a wide variety within your data?
3. Should the process be completely automated or is human interference here and there ok?

Depend on the answers to these questions, I'd inspect the following page:
https://ieeexplore.ieee.org/xpl/conhome/8961318/proceeding

These are the proceedings of the ICDAR 2019 competition, which included a competition for the recognition of tables in PDFs.

Furthermore, although I have not tested it myself, this repository looks pretty good too:
https://github.com/dhavalpotdar/Graph-Convolution-on-Structured-Documents

**Comment by Sharathmk99** [comment]
> 1. Are the invoices of digital nature or are they scanned documents?

Yes its digital PDF documents. Its not scanned.



> 2\. Do the invoices (roughly) follow the same template, or is there a wide variety within your data?

Yes it follows similar structure. Roughly there are 15 to 20 formats.



> 3\. Should the process be completely automated or is human interference here and there ok?

Looking to automate at least 80 to 90 %. We do have human in the loop for correction



> Furthermore, although I have not tested it myself, this repository looks pretty good too:
> https://github.com/dhavalpotdar/Graph-Convolution-on-Structured-Documents

I have tried this way, but some need to explore on inductive vs transductive learning.


Is it possible to use this repository for training on my own data? If your ok we can also have discussion to implement generic framework.

**Comment by rinkstiekema** [comment]
Yes, it is absolutely possible to train the models on your own data. You can view these folders to train your own model:
https://github.com/rinkstiekema/PDF-Table-Structure-Recognition-using-deep-learning/tree/master/pipeline/pix2pixHD
https://github.com/rinkstiekema/PDF-Table-Structure-Recognition-using-deep-learning/tree/master/pipeline/segmentation

Please consult these repositories for the respective training procedure of these models:
https://github.com/NVIDIA/pix2pixHD/
https://github.com/GeorgeSeif/Semantic-Segmentation-Suite

For me it was important to be able to predict images with a somewhat bigger size (1024x1024 pixels). At the time, I could not find any repositories that supported this out of the box. I made some changes to the above mentioned repositories (which you can find in this repository). The pipeline assumes this size and it might be hardcoded in some places. Therefore, I would recommend to use this size and use the forked code. You probably need to pad/crop some of your data. 

Let me know if you have any more questions

**Comment by Sharathmk99** [comment]
Thank you for clarification. I checked the example training dataset from google drive, but i can find png, pdf and csv file.
Based on my understanding to train segmentation model, we need polygon segmentation data right? similar to bounding box right?
Using CSV ground truth data, how can we train segmentation mode i.e pix2pixHD or others.

Please help to understand example dataset to train.. Can i use data from PubTabNet(https://github.com/ibm-aur-nlp/PubTabNet)?

Thank you.
